coda-prototype/src/io/FileUtils.ts

```diff
@@ -184,87 +259,80 @@ class FileUtils {
                         deco_timestamp: boolean = eventRow.hasOwnProperty("deco_timestamp"),
                         deco_author: boolean = eventRow.hasOwnProperty("deco_author");
 
-                    // If this parsed row has the minimum information set required to construct an entry in the dataset,
-                    // construct that entry and add it to the dataset.
-                    // TODO: Break this into smaller functions?
-                    if (id && owner && data) {
-                        if (!dataset) {
-                            dataset = new Dataset(); // TODO: Determine whether this check is necessary.
-                        }
+                    console.assert(id && owner && data);
```

Comment:
This is way too Javascripty for my taste - what is it asserting? non-null?

Reply:
It's possible to parse objects that don't have an id or owner or data, but these shouldn't be included in the Dataset object. This assert replaces an if statement which was here previously (the if was removed by adding code above which filtered out such objects).

If kept, this line would be better placed underneath the filter instruction earlier in the method, but when placed there I don't think it is necessary so I will delete this.